components/map/index.jsx

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Map design pattern](https://lightningdesignsystem.com/components/map/) in React.
+// Based on SLDS v2.4.0
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+
+import Icon from '../icon';
+import Modal from '../modal';
+import { MAP } from '../../utilities/constants';
+
+const displayName = MAP;
+
+const propTypes = {
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Whether the modal containing the map is open. Only for `modal` variant
+	 */
+	isModalOpen: PropTypes.bool,
+	/**
+	 * Title for the Map component.
+	 */
+	title: PropTypes.string,
+	/**
+	 * Accepts a single Map location
+	 */
+	location: PropTypes.string,
+	/**
+	 * Accepts Map locations as an array of Objects
+	 */
+	locations: PropTypes.arrayOf(PropTypes.object),
```

Comment:
This should have object shape.

Reply:
I'd prefer to merge `location` and `locations`. One location would occur by passing in one object into the locations array--that way we don't have multiple data props.

Your keys should be listed out in the prop description comment. See https://github.com/salesforce/design-system-react/blob/master/components/combobox/combobox.jsx#L215